### DISCUSSION TITLE
Are there flags that could potentially improve the image generation time e.g. preload something?

### DISCUSSION THREAD

**Discussion by Sergio-Redraw** [Mikubill/sd-webui-controlnet#1627] [Q&A]
Hello everyone,

I'm new to this forum and I'm working on a project where we are using an API for image generation. We are using ControlNet along with multiple models to achieve the desired results. However, we are facing some challenges regarding the image generation time.

Therefore, I would like to know if there are any specific flags or configurations that we can use to potentially improve the image generation time. For example, are there any preload options that we could explore to optimize the process?

We appreciate any contributions or suggestions you can provide to help us speed up the image generation time in our project.

Thank you in advance.

**Comment by huchenlei** [comment]
You probably want to start a discussion on A1111's repo. https://github.com/AUTOMATIC1111/stable-diffusion-webui/discussions

ControlNet does not offer any performance related options other than `low_vram` and preprocessor result cache (enabled by default).

Have you done any profiling for your whole setup? Is your process's bottleneck in ControlNet?

**Reply by Sergio-Redraw** [reply]
Thanks for sharing the link to A1111's repo. I'll check out the options you mentioned in ControlNet and start a discussion on the repository. ControlNet seems to be one of the bottlenecks in my current process, but I'm aware that there might be other issues as well. I appreciate your suggestions, and I'll work on optimizing my setup, taking into account the performance options provided by ControlNet. If you have any more tips or recommendations, feel free to share. Thank you for your attention and support!
